src/commands/init.js

```diff
@@ -83,84 +222,323 @@ Examples:
+      await toolbox.system.run('git commit -m "Initial commit"', {
+        cwd: directory,
+      })
+      return true
+    }
+  )
+
+const installDependencies = async (toolbox, directory, installCommand) =>
+  await withSpinner(
+    `Install dependencies with ${toolbox.print.colors.muted(installCommand)}`,
+    `Failed to install dependencies`,
+    async spinner => {
+      await toolbox.system.run(installCommand, { cwd: directory })
+      return true
+    }
+  )
+
+const runCodegen = async (toolbox, directory, codegenCommand) =>
+  await withSpinner(
+    `Generate ABI and schema types with ${toolbox.print.colors.muted(codegenCommand)}`,
+    `Failed to generatecode from ABI and GraphQL schema`,
+    async spinner => {
+      await toolbox.system.run(codegenCommand, { cwd: directory })
+      return true
+    }
+  )
+
+const printNextSteps = (toolbox, { subgraphName, directory }, { commands }) => {
+  let { print } = toolbox
+
+  let relativeDir = path.relative(process.cwd(), directory)
+
+  // Print instructions
+  print.success(
+    `
+Subgraph ${print.colors.blue(subgraphName)} created in ${print.colors.blue(relativeDir)}
+`
+  )
+  print.info(`Next steps:
+
+  1. Run \`${print.colors.muted(
+    'graph auth https://api.thegraph.com/deploy/ <access-token>'
+  )}\`
      to authenticate with the hosted service. You can get the access token from
      https://thegraph.com/explorer/dashboard/.
-  
-  2. Type \`cd ${directory}\` to enter the subgraph.
-  
-  3. Run \`${installCommand}\` to install dependencies.
-  
-  4. Run \`${codegenCommand}\` to generate code from contract ABIs and the GraphQL schema.
-  
-  5. Run \`${deployCommand}\` to deploy the subgraph to
+
+  2. Type \`${print.colors.muted(`cd ${relativeDir}`)}\` to enter the subgraph.
+
+  3. Run \`${print.colors.muted(commands.deploy)}\` to deploy the subgraph to
      https://thegraph.com/explorer/subgraph/${subgraphName}.
 
 Make sure to visit the documentation on https://thegraph.com/docs/ for further information.`)
-  },
+}
+
+const initSubgraphFromExample = async (
+  toolbox,
+  { allowSimpleName, subgraphName, directory },
+  { commands }
+) => {
+  let { filesystem, print, system } = toolbox
+
+  // Fail if the subgraph name is invalid
+  if (!revalidateSubgraphName(toolbox, subgraphName, { allowSimpleName })) {
+    process.exitCode = 1
+    return
+  }
+
+  // Fail if the output directory already exists
+  if (filesystem.exists(directory)) {
+    print.error(`Directory or file "${directory}" already exists`)
+    process.exitCode = 1
+    return
+  }
+
+  // Clone the example subgraph repository
+  let cloned = await withSpinner(
+    `Cloning example subgraph`,
+    `Failed to clone example subgraph`,
+    async spinner => {
+      await system.run(
+        `git clone http://github.com/graphprotocol/example-subgraph ${directory}`
+      )
+      return true
+    }
+  )
+  if (!cloned) {
+    process.exitCode = 1
```

Comment:
I tried running `graph init --from-example` without internet connection and it failed silently, probably because of this. We should print something, and maybe also throw an exception so we can also print a more general error?

Reply:
There was a bug in the form processing that would cause it to silently fail when it shouldn't. This should be fixed now.